### ISSUE TITLE
Loss becomes NaN with LEARNING_RATE=0.001

### ISSUE THREAD

**Issue by LegaDyan** [ammirato/target_driven_instance_detection#2]
When I cloned the code and wanted to train the model from the very beginning, I modified the configAVD2 file's LEARNING_RATE to 0.001 and RESIZE_IMG to 1 according to the paper. But this leads to "loss becoming NaN" during the training process after few iterations.

Is there something I've done incorrectly? Thank you so much.
(PyTorch version is 0.4.0)

**Comment by ammirato** [comment]
I'm not sure. When you say it "works on LEARNING_RATE=0.0001" what do you mean by works? You get a loss that isn't NaN? If so, does the network learn anything, i.e. does the mAP improve? Does it work with RESIZE_IMG=1 and LEARNING_RATE=0.0001?  What is the original size of the images?

You can try putting a breakpoint at line 223 of `TDID.py` and see what is the cause of the NaN loss. check the values of all the variables used in lines 221-223.
You can also check on lines 154-158 of `train_tdid.py` to make sure the resizing is happening as you expect.

**Comment by LegaDyan** [comment]
Thank you for the reply.

I trained with the AVD dataset split 2 and the original size is 1920*1080.

Here's the appearance: 
- lr = 0.001 (same with the paper), doesn’t work
- lr = 0.0001
    - resize image to 960*540 (same with the paper), works
    - not resize (same with the original code), doesn’t work
- lr = 0.00001, works whether resize or not.

("not work" stands for I'll get NaN loss and "work" stands for the loss becomes lower and lower, and according to the out_img.jpg, if I train with the lower learning rate, the result makes sense so the network actually learns something. )

And I trained on the setting lr=0.0001 && resize for one epoch and I'm testing it now. Once I have the mAP report I'll let you know.

**Comment by LegaDyan** [comment]
I switch my torch version back to 0.3.1 and won't get this error now.

**Comment by ammirato** [comment]
Ok thanks.

I have experienced similar issues with other code when switching between Pytorch 0.3 and 0.4.

I will make a note on the README, and will continue to look for why the different version causes the issue.

**Comment by LegaDyan** [comment]
Thank you so much

**Comment by asdfqwer2015** [comment]
@Qiaoyf96 @ammirato 
FYI, I've met the similar issue, and after debug, it solved.
In `TDID.py`, `build_loss` method of `TDID` class,
the line `loss_box = F.smooth_l1_loss(bbox_pred, bbox_targets, size_average=False) / (fg_cnt + 1e-4)`
=>
`loss_box = F.smooth_l1_loss(bbox_pred, bbox_targets, size_average=False) / (fg_cnt.float() + 1e-4)`
`fg_cnt` seems to be int type before, and which makes `1e-4` term fails to regularize the formula.
